src/model_training.py

```diff
@@ -312,8 +312,13 @@ def convert_feature_dataframe_to_dict(
 test_gen = DataGenerator(train_data=test_data.limit(10))
 # load input data table as a Spark DataFrame
 input_data = test_gen.training_set.load_df().toPandas()
+# Merge to weather file city index 
+input_data_weather_file_city_index = (
+    input_data
+    .merge(test_gen.weather_features_df[['weather_file_city', 'weather_file_city_index']], on = 'weather_file_city', how = 'left')
```

Comment:
This is a bit janky - ideally we want the `weather_file_city_index` to be included in the `test_gen.training_set` DF, but there isn't an easy way to do this such that the city <> index is the same as the `weather_features_df`

Reply:
The most import key to all of this, however it is done, is that once an index has been assigned to a weather city it can never change for the lifetime of the model, no matter how or where inference is done.

Reply:
Also I think that as is, this join to grab the weather city index would always have to be applied at inference time. This means we would need to apply this fix somewhere [here](https://github.com/rewiringamerica/surrogate_modeling/blob/662bb2eb4a96fd1d7b8724e3236bbe06b5da7416/src/model_evaluation.py#L90) too, right? Not to mention by the caller at inference time inside dohyo as well, which does not seem ideal. 

As a result of the additional code needed in `model_evaluation.py`, I'm gonna consider this a blocker. I think the suggestion I made will address this issue tho.

Reply:
I've added the `weather_file_index` to the `build_feature_store.py` so it will be included in all training sets and doesn't have to be subsequently merged at the batch creation.